client/src/features/project/components/ProjectSettingAvatar.tsx

```diff
@@ -0,0 +1,195 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useCallback, useEffect, useState } from "react";
+import { useGetProjectIndexingStatusQuery } from "../projectKgApi";
+import {
+  useProjectMetadataQuery,
+  useUpdateAvatarProjectMutation,
+} from "../../projects/projectsKgApi";
+import { SettingRequiresKg } from "./ProjectSettingsUtils";
+import {
+  extractRkErrorRemoteBranch,
+  RtkErrorAlert,
+} from "../../../components/errors/RtkErrorAlert";
+import ProjectWarningForMerge from "./ProjectWarningForMerge";
+import { InputCard } from "../../../components/inlineSubmitInput/InlineSubmitInput";
+import { PROJECT_AVATAR_MAX_SIZE } from "../../../project/new/components/NewProjectAvatar";
+import InlineSubmitImageInput, {
+  ImageValue,
+  INITIAL_IMAGE_VALUE,
+} from "../../../components/inlineSubmitImageInput/InlineSubmitImageInput";
+import { ImageFieldPropertyName as Prop } from "../../../components/form-field/FormGeneratorImageInput";
+import { Card, CardBody } from "reactstrap";
+import { ImagesLinks } from "../Project";
+import { getEntityImageUrl } from "../../../utils/helpers/HelperFunctions";
+
+const CURRENT_AVATAR_NAME = "[Current Avatar]";
+
+interface ProjectSettingsAvatarProps {
+  isMaintainer: boolean;
+  projectFullPath: string;
+  projectId: number;
+  branch?: string;
+  gitUrl: string;
+}
+export function ProjectSettingsAvatar({
+  isMaintainer,
+  projectFullPath,
+  projectId,
+  branch,
+  gitUrl,
+}: ProjectSettingsAvatarProps) {
+  const [avatar, setAvatar] = useState<ImageValue>();
+  const [succeeded, setSucceeded] = React.useState<boolean | undefined>(
+    undefined
+  );
+  const projectIndexingStatus = useGetProjectIndexingStatusQuery(projectId, {
+    skip: !projectFullPath || !projectId,
+  });
+  const projectMetadata = useProjectMetadataQuery(
+    { projectPath: projectFullPath, projectId },
+    {
+      skip:
+        !projectFullPath ||
+        !projectId ||
+        !projectIndexingStatus.data?.activated,
+    }
+  );
+
+  const [
+    updateProject,
+    {
+      isLoading: isLoadingMutation,
+      isSuccess,
+      isError,
+      error: errorAvatar,
+      reset,
+    },
+  ] = useUpdateAvatarProjectMutation();
+
+  const onSubmit = useCallback(() => {
+    const imageProject = avatar?.options[avatar?.selected];
+    if (!imageProject) return;
+    updateProject({
+      projectPathWithNamespace: projectFullPath,
+      avatar: imageProject.FILE as File,
+      projectId,
+    })
+      .unwrap()
+      .then(() => setSucceeded(true))
+      .catch(() => setSucceeded(false));
+  }, [avatar, projectFullPath, updateProject, projectId]);
+
+  const setAvatarAndReset = (newAvatar: ImageValue) => {
+    setAvatar(newAvatar);
+    reset();
+  };
```

Comment:
I think useCallback should be used here...
```suggestion
  const setAvatarAndReset = useCallback((newAvatar: ImageValue) => {
    setAvatar(newAvatar);
    reset();
  }, [setAvatar, reset]);
```